backend/Dockerfile

```diff
@@ -12,47 +12,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Build arguments
-ARG SOURCE_CODE=.
+# 1. Build api server application
+FROM golang:1.21.7-bookworm as builder
+RUN apt-get update && apt-get install -y cmake clang musl-dev openssl
+WORKDIR /go/src/github.com/kubeflow/pipelines
```

Comment:
this path needs to be pointing to where the odh/dsp APIServer code is, and not kubeflow/pipelines, I believe.

That might be causing this CI failure: https://github.com/opendatahub-io/data-science-pipelines/actions/runs/11960604493/job/33344984082?pr=119

```
Error from server (BadRequest): container "ml-pipeline-api-server" in pod "ml-pipeline-6cc48b7b9-zc785" is waiting to start: trying and failing to pull image
```

It's looking for an `ml-pipeline-` apiserver pod, which we find in a KFP deployment. Corresponding APIServer pod in DSP would have `ds-pipelines-` prefix.

Reply:
@DharmitD @VaniHaripriya 

From https://github.com/opendatahub-io/data-science-pipelines/pull/119/files#r1852961305:
> We want to leave the upstream Dockerfiles unchanged to avoid merge conflicts during rebase. Instead, we relocated the DSP Dockerfiles to the odh_images folder

That said, it seems like we should leave the original Dockerfiles identical to upstream.
If it's causing any errors, we may need to figure out how to work around these errors without having to change the Dockerfiles that come from upstream.

Does my point of view make sense?